src/DynamoCore/DSEngine/ExecutionInstance.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Dynamo.Core.IPC;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+
+namespace Dynamo.DSEngine
+{
+    /// <summary>
+    /// ExecutionInstance's job is to run the graph.
+    /// </summary>
+    class ExecutionInstance
+    {
+        private Queue<Message> messageQueue = new Queue<Message>();
+        private Thread messageThread;
```

Comment:
Since this is a receiving end, I thought about it for a while, is it possible to have this end receives and recreate AST objects? If that's all it needs, then we may not even want an instance of EC here. `LiveRunner` instance is all that should be owned by `ExecutionInstance`.